src/sage/homology/homology_group.py

```diff
@@ -114,7 +114,7 @@ def _latex_(self):
         eldv = self._original_invts
         if len(eldv) == 0:
             return "0"
-        rank = len([x for x in eldv if x == 0])
+        rank = len([1 for x in eldv if x == 0])
```

Comment:
smaller use of memory

Reply:
hm... maybe you meant to write `sum(1 for ...)`?

Reply:
well, this would be another option. Not clear to me which is the best. Could depend on the content type of the list ?

Reply:
`[x for x...]` and `[1 for x...]` use the same amount of memory. No item is copied. It does not matter what the items are. Only a list is consed.

`sum([1 for ...]) also uses the same amount of memory because it first constructs the list.

`sum(1 for ...)` uses no memory.

I don't know which of the options is faster, haven't tried.

Not that any of this matters...